nexus/test-utils/src/lib.rs

```diff
@@ -1716,3 +1717,40 @@ pub async fn start_dns_server(
 
     Ok((dns_server, http_server, resolver))
 }
+
+/// Wait until a producer is registered with Oximeter.
```

Comment:
FYI: This was moved as-is, with no changes, from nexus/tests/integration_tests/metrics.rs